src/Refinement/RefinedDiscreteModels.jl

```diff
@@ -0,0 +1,116 @@
+
+
+"""
+Refinement glue between two nested triangulations
+- `f2c_faces_map`     : Given a fine face gid, returns the gid of the coarse face containing it. 
+- `fcell_to_child_id` : Given a fine cell gid, returns the local child id within the coarse cell containing it.
+- `f2c_ref_cell_map`  : Ref coordinate map between the fcells and their parent ccell, i.e 
+                          for each fine cell defines Φ st. x_c = Φ(x_f)
+"""
+struct RefinementGlue{A,B,C} <: GridapType
+  f2c_faces_map::A
+  fcell_to_child_id::B
+  f2c_ref_cell_map::C
+
+  function RefinementGlue(f2c_faces_map,
```

Comment:
remove inner constructor. as now defined it is equivalent to the default inner constructor

Reply:
This inner constructor was there because `f2c_ref_cell_map` used to be computed from the other elements. However I now compute it outside to give more flexibility. I can remove the constructor.

Reply:
However I might keep it, since I am thinking of precomputing the inverse map `ccell_to_fcell`. Right now it is computed every time you apply the operator in the function: 

```
function Gridap.Refinement.f2c_cell_contrs(trian::RefinedTriangulation{Dc,Dp},cell_vec) where {Dc,Dp}
  @check num_cells(trian) == length(cell_vec)

  model = get_refined_model(trian)
  glue  = get_glue(model)
  nF    = num_cells(trian)
  nC    = num_cells(get_parent(model))
  nChildren = 4 # Num fine cells per coarse cell

  # Invert fcell_to_ccell
  fcell_to_ccell = glue.f2c_faces_map[Dc+1]
  ccell_to_fcell = [fill(-1,nChildren) for i in 1:nC]
  cidx = fill(1,nC)
  for iF in 1:nF
    iC = fcell_to_ccell[iF]
    ccell_to_fcell[iC][cidx[iC]] = iF
    cidx[iC] += 1
  end

  # Map that sums fine contributions for each coarse cell
  return lazy_map((I,V)->sum(V[I]),ccell_to_fcell,Fill(cell_vec,nC))
end
```
However I am still thinking on how I can further optimize this function (both in generality and allocations), so don't take it as a final version.

Reply:
In particular, `nChildren` needs to be replaced by a non-uniform number of children counts for each cell (which is necessary for non-uniform refinement, mixed refinement+coarsening, mixed-element meshes and so on...

Reply:
I'll probably think about it today

Reply:
Resolved in [356ddd7](https://github.com/gridap/Gridap.jl/pull/838/commits/356ddd730acd280518edec7686d97822beab7711)